src/msg_recorder/replayer.cc

```diff
@@ -19,7 +19,25 @@ void MessageReplayer::SetSpeedupRate(double rate) {
     speed_up_rate_ = rate;
 }
 
+void MessageReplayer::SetStartCallback(std::function<void()>&& on_start) {
+    on_start_ = std::move(on_start);
```

Comment:
1. I still think setter is overkill.
2. Better to name with prexxx/postxxx for state xxx in state machine, or names will start get confusing later on.